Data/PackageData.cs

```diff
@@ -404,12 +403,11 @@ private static async Task<PackageData> ReadPackageFromUrl (PackageData package,
             {
                 //System.Console.WriteLine($"DOWNLOADING {package.DownloadUrl}");
                 var r = await httpClient.GetAsync (package.DownloadUrl, token).ConfigureAwait (false);
-                var data = new MemoryStream ();
-                using (var s = await r.Content.ReadAsStreamAsync ().ConfigureAwait (false)) {
-                    await s.CopyToAsync (data, 16 * 1024, token).ConfigureAwait (false);
+                using (var content = r.Content) {
+                    using (var s = await content.ReadAsStreamAsync ().ConfigureAwait (false)) {
+                        await Task.Run (() => package.Read (s, httpClient), token).ConfigureAwait (false);
```

Comment:
What's the `Task.Run` for?

Reply:
Not sure, that was already here (but further below) so I didn't want to change it. Personally I don't think it gains anything (apart from changing the thread that is doing the processing)

Reply:
Unless I'm missing something, this does not avoid blocking a threadpool thread. See: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#avoid-using-taskrun-for-long-running-work-that-blocks-the-thread

Reply:
I've got no problem with that being removed, but that's what the code looked like originally (I just moved this up from line 412) and it's used in a large number of other places in the codebase.

Having said that, I'm not sure that what you've linked to applies here though because it states that
> Long running work in this context refers to a thread that's running for the lifetime of the application doing background work

This line isn't long running, and it's waited on before the handling of the request continues.

I would consider changing this line to be better suited to another pull request which deals with the other occurrences too

Reply:
> I would consider changing this line to be better suited to another pull request which deals with the other occurrences too

Agreed 👍

>  ... I'm not sure that what you've linked to applies here though ...

My guess is that the `Task.Run` was added as the `Read` method may block for a long time for certain packages. If so, this bit is relevant:

> Task.Run will queue a work item to the thread pool ... Instead, spawn a new thread manually to do long running blocking work.

The thread pool also processes requests. In other words, the `Task.Run` does not prevent long running `Read`s from starving all other requests. One way to mitigate this problem would be to create dedicated threads for the `Read` operations to reduce the impact of thread starvation. But like you said, this is out-of-scope for your change